packages/compiler/src/types.ts

```diff
@@ -91,13 +94,52 @@ export interface ImportExpression {
   reference?: string;
 }
 
+interface WithBlockHeight {
+  [BLOCK_HEIGHT_KEY]: number;
+}
+
+interface SourceWithBlockHeight extends WithBlockHeight {
+  "": string;
+}
+
 export interface ComponentEntry {
   '': string;
   css: string;
 }
 
+export interface ComponentEntryWithBlockHeight extends WithBlockHeight {
+  "": SourceWithBlockHeight,
+  css: SourceWithBlockHeight;
+}
+
+interface SocialWidget {
+  [name: string]: ComponentEntry
+}
+
+export interface SocialWidgetWithBlockHeight {
+  [name: string]: ComponentEntryWithBlockHeight | number;
+}
+
 interface SocialComponent {
-  widget: { [name: string]: string | ComponentEntry };
+  widget: SocialWidget;
 }
 
-export type SocialComponentsByAuthor = { [author: string]: SocialComponent };
+export interface SocialComponentWithBlockHeight extends WithBlockHeight, SocialWidgetWithBlockHeight { }
+
+export interface SocialComponentsByAuthor {
+  [author: string]: SocialComponent
+};
+
+export interface SocialComponentsByAuthorWithBlockHeight {
+  [authorOrBlockHeight: string]: SocialComponentWithBlockHeight | number;
+}
+
+export type ComponentSourcesResponse = { [key: string]: BOSModule }
+
+export interface ComponentCacheRecord {
+  /* <componentPath>@<blockHeight> */
```

Comment:
I think we should exclude this comment - at best it's a hint at how the type is used but at worst it's outdated and misleading when the usage changes